deploy/tests/000037-legacy-staking-rewards-claim.ts

```diff
@@ -0,0 +1,108 @@
+    let networkInfo: BancorNetworkInfo;
+    let stakingRewardsClaim: StakingRewardsClaim;
+    let merkleTree: MerkleTree;
+
+    before(async () => {
+        merkleTree = new MerkleTree(
+            Object.entries(snapshot).map(([provider, { claimable }]) => generateLeaf(provider, claimable)),
+            keccak256,
+            { sortPairs: true }
+        );
+    });
+
+    beforeEach(async () => {
+        bnt = await DeployedContracts.BNT.deployed();
+        bnBNT = await DeployedContracts.bnBNT.deployed();
+        networkInfo = await DeployedContracts.BancorNetworkInfo.deployed();
+        stakingRewardsClaim = await DeployedContracts.StakingRewardsClaim.deployed();
+    });
+
+    const generateLeaf = (address: string, amount: string) =>
+        Buffer.from(solidityKeccak256(['address', 'uint256'], [getAddress(address), amount]).slice(2), 'hex');
+
+    it('should deploy the legacy staking rewards claim contract', async () => {
+        expect(await stakingRewardsClaim.merkleRoot()).to.equal(merkleRoot);
+
+        const standardRewards = await DeployedContracts.StandardRewards.deployed();
+        const bntPool = await DeployedContracts.BNTPool.deployed();
+        const bntGovernance = await DeployedContracts.BNTGovernance.deployed();
+        const liquidityProtection = await DeployedContracts.LiquidityProtection.deployed();
+
+        const expectedRoles = isMainnet()
+            ? [standardRewards.address, bntPool.address, liquidityProtection.address, stakingRewardsClaim.address]
+            : [standardRewards.address, bntPool.address];
+        await expectRoleMembers(
+            bntGovernance as any as AccessControlEnumerable,
+            Roles.TokenGovernance.ROLE_MINTER,
+            expectedRoles
+        );
+    });
+
+    const testClaim = (stake: boolean, providerIndices: [number, number]) => {
+        it(`should allow ${stake ? 'staking' : 'claiming'} legacy staking rewards`, async () => {
+            const sampleProviders = Object.fromEntries(Object.entries(snapshot).slice(...providerIndices));
+
+            for (const [provider, { claimable: amount }] of Object.entries(sampleProviders)) {
+                if (BigNumber.from(amount).isZero()) {
+                    continue;
+                }
+
+                const signer = await ethers.getSigner(provider);
+                const proof = merkleTree.getHexProof(generateLeaf(provider, amount));
+
+                const prevBNTTotalSupply = await bnt.totalSupply();
+                const prevBNTBalance = await bnt.balanceOf(provider);
+                const prevBNBNTBalance = await bnBNT.balanceOf(provider);
+
+                const poolTokenAmount = await networkInfo.underlyingToPoolToken(bnt.address, amount);
+
+                const method = stake
+                    ? stakingRewardsClaim.connect(signer).stakeRewards
+                    : stakingRewardsClaim.connect(signer).claimRewards;
+                await method(provider, amount, proof);
```

Comment:
Minor - can simplify by replacing this:
```
const signer = await ethers.getSigner(provider);
```
With this:
```
const xxx = stakingRewardsClaim.connect(await ethers.getSigner(provider));
const method = stake ? xxx.stakeRewards : xxx.claimRewards;
```

Reply:
re: `await ethers.getSigner(provider)` - we're not going to "compress" code if it makes it less readable 
re: DRYing `connect` - sure